### PR TITLE
add query/invoke strategy settings to give flexibilty

### PR DESCRIPTION
**Pull request by inalgnu**

### CODE REVIEW

File: backend/substrapp/ledger_utils.py [hunk 1]
```diff
@@ -154,6 +154,23 @@ def get_hfc():
...
 
+def get_invoke_signing_peers(current_peer, all_peers):
+    return _get_signing_peers(strategy=settings.LEDGER_INVOKE_STRATEGY, current_peer=current_peer, all_peers=all_peers)
+
+
+def get_query_singing_peers(current_peer, all_peers):
+    return _get_signing_peers(strategy=settings.LEDGER_QUERY_STRATEGY, current_peer=current_peer, all_peers=all_peers)
+
+
+def _get_signing_peers(strategy, current_peer, all_peers):
```
Comment (samlesu) [inline backend/substrapp/ledger_utils.py]: Minor, it looks like we are defining low to high level methods from top to bottom.
So, here `_get_signing_peers()` could be defined before `get_invoke_signing_peers` / `get_query_singing_peers`

Comment (inalgnu) [reply backend/substrapp/ledger_utils.py]: done 👍

Comment (Kelvin-M) [reply backend/substrapp/ledger_utils.py]: I think it's better to talk about `endorsing` peers than `signing` peers :

> Endorsing peer is the peer that will "endorse"/provide the seal of approval to a transaction when it is proposed. After the transaction is endorsed, the transaction(plus the endorsement) will be submitted to blockchain.

File: backend/substrapp/ledger_utils.py [hunk 2]
```diff
@@ -154,6 +154,23 @@ def get_hfc():
         loop.close()
 
 
+def _get_signing_peers(strategy, current_peer, all_peers):
+    if strategy == 'SELF':
+        return [current_peer]
+    if strategy == 'MAJORITY':
```
Comment (Kelvin-M) [inline backend/substrapp/ledger_utils.py]: This looks more like `ALL `  than `MAJORITY`.
All Majority combinations will be constructed by   
```python
from itertools import combinations 
all_comb = combinations(all_peers, len(all_peers)//2 + 1) 

```

File: backend/substrapp/ledger_utils.py (hunk 1 above)
Comment (inalgnu) [reply backend/substrapp/ledger_utils.py]: done here 9e65ccc

File: backend/substrapp/ledger_utils.py (hunk 2 above)
Comment (inalgnu) [reply backend/substrapp/ledger_utils.py]: Thank's @Kelvin-M, fixed here 3a9d353

**Pull request merged**